src/attributes/classes.js

```diff
@@ -114,22 +114,31 @@ jQuery.fn.extend({
 		}
 
 		return this.each(function() {
-			var className, i, self, classNames;
+			var className, i, self, classNames, predicate;
 
 			if ( type === "string" ) {
 
 				// Toggle individual class names
 				i = 0;
 				self = jQuery( this );
+				predicate = jQuery.isFunction( stateVal ) ? stateVal.call( self, i, self) : null;
```

Comment:
I didn't exactly understood this line but normally jQuery send the DOM element itself, not a jQuery object.
And `i` will be always `0` no?